crates/client/src/client/types/translation.rs

```diff
@@ -0,0 +1,191 @@
+use crate::client::{
+    pagination::PaginatedResult,
+    schema,
+    types::{
+        block::{
+            Genesis,
+            Header,
+            PoAConsensus,
+        },
+        message::{
+            Message,
+            MessageProof,
+        },
+        primitives::{
+            Bytes32,
+            Bytes64,
+            BytesN,
+        },
+        scalars::TransactionId,
+        Block,
+        Consensus,
+        MerkleProof,
+    },
+};
+
+impl From<schema::block::Header> for Header {
+    fn from(value: schema::block::Header) -> Self {
+        let id: Bytes32 = value.id.0 .0.into();
+        let transactions_root: Bytes32 = value.transactions_root.0 .0.into();
+        let message_receipt_root: Bytes32 = value.message_receipt_root.0 .0.into();
+        let prev_root: Bytes32 = value.prev_root.0 .0.into();
+        let application_hash: Bytes32 = value.application_hash.0 .0.into();
+
+        Self {
+            id: id.into(),
+            da_height: value.da_height.0.into(),
+            transactions_count: value.transactions_count.0,
+            message_receipt_count: value.message_receipt_count.0,
+            transactions_root: transactions_root.into(),
+            message_receipt_root: message_receipt_root.into(),
+            height: value.height.0.into(),
+            prev_root: prev_root.into(),
+            time: value.time.0.into(),
+            application_hash,
+        }
+    }
+}
+
+impl From<schema::block::Consensus> for Consensus {
```

Comment:
I'd probably just put these translation types into the mod that defines the type they convert into. In this case the `block` mod